scripts/ConvertToAckermann.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+"""
+Subscribes to a twist message and converts it to an ackermann message.
+Subscribes to /twist_vel
+Publishes to /cmd_vel
+
+Assumes all inputs are normalized between -1 and 1
+
+@edited: 9/28/18
+@author: Amy Phung
+"""
+
+import rospy
+from geometry_msgs.msg import Twist
+from ackermann_msgs.msg import AckermannDrive
+from scipy.interpolate import interp1d
+
+class ConvertToAckermann():
+    def __init__(self):
+        # Define variables
+        self.twist_data = None
+
+        # Define ROS constructs
+        rospy.init_node("convert_to_ackermann")
+        self.twist_sub = rospy.Subscriber("/twist_vel", Twist, self.twist_cb)
+        self.ack_pub = rospy.Publisher("/drive", AckermannDrive, queue_size=10)
+        self.update_rate = rospy.Rate(10)
+
+    def twist_cb(self,msg):
+        self.twist_data = msg
+
+    def twist_to_ackermann(self,linear_vel, angular_vel):
+        """
+        Converts linear and angular velocities to linear velocity and steering angle for
+        ackermann messages
+
+        Args:
+            linear_vel - forward linear velocity from Twist message (should be between -1 and 1)
+            angular_vel - angular velocity from Twist message (should be between -1 and 1)
+        """
+        # Assume twist message is angular vel from -1 to 1, velocity is -1 to 1
+        # steering in degrees from -45 to 45, velocity is -2 to 2
+        vel_scale = interp1d([-1,1],[-2,2])
+        angle_scale = interp1d([-1,1],[-45,45])
+
+        ack_msg = AckermannDrive()
+        ack_msg.speed = vel_scale(linear_vel)
+        ack_msg.steering_angle = angle_scale(angular_vel)
+
+        return ack_msg
+
+    def run(self):
+        # Takes no args, executes timed loop for node
+        while (self.twist_data == None) and not rospy.is_shutdown():
+            print('MSG: No twist data published')
```

Comment:
Use `rospy.loginfo("...")`

Reply:
It is easier and clearer to  merge the two while loops by deleting the one above and inserting this in the other while loop

```python
if self.twist_data == None:
    rospy.loginfo("...")
    continue